### ISSUE TITLE
`predictions` is an object when predicting over a single sample, an array when predicting over many

### ISSUE THREAD

**Issue by nrinaudo** [beniz/deepdetect#177]
Ok, this one is a bit hard to describe but very clear with an example.

When submitting an array of a single inline document for prediction, DeepDetect's answer will not contain a JSON array, but a JSON object. This makes client-side parsing rather cumbersome.

Here's a concrete example:

Request:
```
curl -XPOST "http://localhost:8080/predict" -d "{\"service\":\"foobar\",\"data\":[\"0:-1.15554170033642 1:-1.323809487778449"]}"
```

Response:
```json
{  
   "status":{  
      "code":200,
      "msg":"OK"
   },
   "head":{  
      "method":"/predict",
      "service":"foobar",
      "time":9.0
   },
   "body":{  
      "predictions":{  
         "uri":"0",
         "classes":{  
            "last":true,
            "prob":0.8474653959274292,
            "cat":"1"
         }
      }
   }
}
```

Note how `body.predictions` is an object rather than an array.

**Comment by beniz** [comment]
This should be fixed by PR #182

**Comment by beniz** [comment]
Fixed by merged PR #182 that led to an overall long-due improvement of the internal data structures. Thanks.
